packages/jest-haste-map/src/ModuleMap.ts

```diff
@@ -3,52 +3,49 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
- *
- * @flow
  */
 
-import type {Path} from 'types/Config';
-import type {
+import {Config} from '@jest/types';
+import {
   DuplicatesSet,
   HTypeValue,
   ModuleMetaData,
   RawModuleMap,
   ModuleMapData,
   DuplicatesIndex,
   MockData,
-} from 'types/HasteMap';
+} from './types';
 
 import * as fastPath from './lib/fast_path';
 import H from './constants';
 
-const EMPTY_OBJ = {};
+const EMPTY_OBJ = {} as {[key: string]: any};
 const EMPTY_MAP = new Map();
 
-export opaque type SerializableModuleMap = {
+type ValueType<T> = T extends Map<T, infer V> ? V : T;
+
+export type SerializableModuleMap = {
   // There is no easier way to extract the type of the entries of a Map
-  duplicates: $Call<
-    typeof Array.from,
-    $Call<$PropertyType<DuplicatesIndex, 'entries'>>,
-  >,
-  map: $Call<typeof Array.from, $Call<$PropertyType<ModuleMapData, 'entries'>>>,
-  mocks: $Call<typeof Array.from, $Call<$PropertyType<MockData, 'entries'>>>,
-  rootDir: string,
+  duplicates: [string, ValueType<DuplicatesIndex>];
+  map: [string, ValueType<ModuleMapData>];
+  mocks: [string, ValueType<MockData>];
+  rootDir: Config.Path;
 };
 
 export default class ModuleMap {
-  _raw: RawModuleMap;
-  static DuplicateHasteCandidatesError: Class<DuplicateHasteCandidatesError>;
+  private readonly _raw: RawModuleMap;
+  static DuplicateHasteCandidatesError: DuplicateHasteCandidatesError;
 
   constructor(raw: RawModuleMap) {
     this._raw = raw;
   }
 
   getModule(
     name: string,
-    platform: ?string,
-    supportsNativePlatform: ?boolean,
-    type: ?HTypeValue,
-  ): ?Path {
+    platform: string | null,
+    supportsNativePlatform: boolean | null,
+    type: HTypeValue | null,
+  ): string | null {
     if (!type) {
```

Comment:
If we're sure only falsy value is null here, let's be explicit about it to make check faster :)